src/main/java/org/ballerinalang/plugins/idea/psi/impl/BallerinaPsiImplUtil.java

```diff
@@ -1587,50 +1587,66 @@ public static PsiElement resolveElementInScope(@NotNull IdentifierPSINode identi
                                                    boolean matchConstants, boolean matchEndpoint) {
         ScopeNode scope = PsiTreeUtil.getParentOfType(identifier, CodeBlockScope.class, VariableContainer.class,
                 TopLevelDefinition.class, LowerLevelDefinition.class);
-        if (scope != null) {
-            int caretOffset = identifier.getStartOffset();
-            if (matchLocalVariables) {
-                List<IdentifierPSINode> variables = BallerinaPsiImplUtil.getAllLocalVariablesInResolvableScope(scope,
-                        caretOffset);
-                for (IdentifierPSINode variable : variables) {
-                    if (identifier.getText().equals(variable.getText())) {
-                        return variable;
-                    }
+        if (scope == null) {
+            return null;
+        }
+        int caretOffset = identifier.getStartOffset();
+        if (matchLocalVariables) {
+            List<IdentifierPSINode> variables = BallerinaPsiImplUtil.getAllLocalVariablesInResolvableScope(scope,
+                    caretOffset);
+            for (IdentifierPSINode variable : variables) {
+                if (variable == null) {
+                    continue;
+                }
+                if (identifier.getText().equals(variable.getText())) {
+                    return variable;
                 }
```

Comment:
Shall we refactor this method by removing duplicated for loop here? List<IdentifierPSINode> can be initialized at the top and then iterate it